### ISSUE TITLE
Documentation fixes - how to proceed?

### ISSUE THREAD

**Issue by ecs-hk** [NYCComptroller/Checkbook#47]
After deploying Checkbook NYC repeatedly over the last few days (on Amazon EC2 and RDS), I have found a number of issues in the current INSTALL.md and SOLR-INSTALL.md.

# Show stoppers

Some of the documentation issues are complete show stoppers. For instance, this one:

> $ sudo su www-data
> $ cp -a source/webapp/* /var/www/html

Following these steps will result in a Drupal installation that doesn't work at all. The `/*` shell glob does not capture dotfiles (e.g. `.htaccess`), so the URI rewrites will not work properly.

# Security suggestions

Other documentation issues are simply suggested changes for improved security. In my internal notes, I've modified the steps to create a service account for Apache Tomcat, and run it with the unprivileged account. Similarly, the Drupal files are root-owned and read-only (with a handful of exceptions) by the Apache HTTP Server service account.

# Question

I'd like to submit a pull request to make various documentation changes. Is it OK if I bundle it all up as a single pull request? Or is it cleaner to first fix "show stopper" problems and then a second pull request for "security suggestions"?

**Comment by kfogel** [comment]
First, thanks for making doc improvements!  I think it's better to have a a few separate changes, rather than bundle it all together.  It makes it easier to review, and easier for someone looking at the history later to understand the purpose of each change.  Disclaimer: I'm mainly an observer here, not a member of the Checkbook dev team proper.   But I've worked with them a bit and can push changes, so I'll review your doc PRs as best I can and either merge them or try to find the right person to review them if I'm not knowledgeable enough to do so.  Most likely I'll be able to merge them, from your description above.

**Comment by ecs-hk** [comment]
Of course. I am happy to provide feedback and pull requests. One question before I start submitting documentation-related pull requests:

Is there a particular naming I should follow for pull requests? I noticed there are "hotfix_xyz" branches that have been merged into master. I've forked NYCComptroller/Checkbook, and it occurs to me that you may not want me submitting pull requests from the forked project's master branch.

**Comment by kfogel** [comment]
Oh, it's fine to submit from the forked repository's master branch if that's easier for you.  But if you want to create branches in your repository, use whatever naming convention works for you.  Personally for doc fixes I usually use `doc_foo` or `doc_foo_N` (where "N" is some issue number, if there is an associated issue).  But I think branch namespace management is pretty much per-repository -- while the devs in the CheckbookNYC main repository seem to be using some conventions, those conventions don't necessarily apply outside that repository anyway.

**Comment by ecs-hk** [comment]
Karl, thanks for the suggestions - and for being patient. I am finally re-engaging in working on this.

I've submitted pull request #49 to address what I consider to be "show stopper" issues. (One is failing to mention that .htaccess must be copied, which breaks Checkbook. Another is the pear/drush instructions no longer work.)

If these look OK to be merged, I will get started on a couple changes for improved security in the installation.

**Comment by kfogel** [comment]
Will take a look in #49 -- thanks, @ecs-hk!
